process-engine/process_engine
# send error during process execution

## What did you change?

I adjusted the error handling to send an extra message on the process instance channel as described in https://github.com/process-engine/process_engine/issues/43.

The process engine subscribes to that channel when executing a process and rejects if it receives an error message.

## How can others test the changes?

Execute a process with two service tasks via the `ProcessEngineService` and throw an error in the method executed by the first service task. The second service task will not be executed, the promise of the process execution resolves and the error stays hidden (it only gets assigned to the process token as the result of the task producing the error).

Edit: Now the error serialization as well as typed errors are working together with the changes of this PR. As a consequence this PR relies on https://github.com/process-engine/process_engine_contracts/pull/16, which should be merged **BEFORE** this PR gets merged.

## PR-Checklist

Please check the boxes in this list after submitting your PR:
